docs/src/man/basics.md

```diff
@@ -0,0 +1,1505 @@
+columns.
+
+```jldoctest dataframe
+julia> show(german, allcols=true)
+1000×10 DataFrame
+  Row │ id     Age    Sex     Job    Housing  Saving accounts  Checking account  Credit amount  Duration  Purpose
+      │ Int64  Int64  String  Int64  String   String           String            Int64          Int64     String
+──────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+    1 │     0     67  male        2  own      NA               little                     1169         6  radio/TV
+    2 │     1     22  female      2  own      little           moderate                   5951        48  radio/TV
+    3 │     2     49  male        1  own      little           NA                         2096        12  education
+    4 │     3     45  male        2  free     little           little                     7882        42  furniture/equipment
+    5 │     4     53  male        2  free     little           little                     4870        24  car
+    6 │     5     35  male        1  free     NA               NA                         9055        36  education
+    7 │     6     53  male        2  own      quite rich       NA                         2835        24  furniture/equipment
+    8 │     7     35  male        3  rent     little           moderate                   6948        36  car
+  ⋮   │   ⋮      ⋮      ⋮       ⋮       ⋮            ⋮                ⋮                ⋮           ⋮               ⋮
+  994 │   993     30  male        3  own      little           little                     3959        36  furniture/equipment
+  995 │   994     50  male        2  own      NA               NA                         2390        12  car
+  996 │   995     31  female      1  own      little           NA                         1736        12  furniture/equipment
+  997 │   996     40  male        3  own      little           little                     3857        30  car
+  998 │   997     38  male        2  own      little           NA                          804        12  radio/TV
+  999 │   998     23  male        2  free     little           little                     1845        45  radio/TV
+ 1000 │   999     27  male        2  own      moderate         moderate                   4576        45  car
+                                                                                                              985 rows omitted
+```
+
+You can also compute descriptive statistics directly on individual columns:
+
+```jldoctest dataframe
+julia> using Statistics
+
+julia> mean(german.Age)
+35.546
+```
+
+`mapcols` function return a `DataFrame` where each column of `german` is transformed using function `f`.
+Note that `mapcols` guarantees not to reuse the columns from `german` in the returned `DataFrame`. If `f` 
+returns its argument then it gets copied before being stored. 
+
+```jldoctest dataframe
+julia> german = copy(german_ref);
```

Comment:
why is this line needed?

Reply:
because here I needed the original data frame to perform the further operation.

Reply:
but was it modified at this point?